query/src/pipelines/new/processors/transforms/aggregator/aggregator_single_key.rs

```diff
@@ -54,26 +54,31 @@ pub struct SingleStateAggregator<const FINAL: bool> {
 impl<const FINAL: bool> SingleStateAggregator<FINAL> {
     pub fn try_create(params: &Arc<AggregatorParams>) -> Result<Self> {
         let arena = Bump::new();
-        let (layout, offsets_aggregate_states) =
-            unsafe { get_layout_offsets(&params.aggregate_functions) };
-
-        let get_places = || -> Vec<StateAddr> {
-            let place: StateAddr = arena.alloc_layout(layout).into();
-            params
-                .aggregate_functions
-                .iter()
-                .enumerate()
-                .map(|(idx, func)| {
-                    let arg_place = place.next(offsets_aggregate_states[idx]);
-                    func.init_state(arg_place);
-                    arg_place
-                })
-                .collect()
-        };
-
-        let places = get_places();
-        let temp_places = get_places();
 
+        let mut places: Vec<StateAddr> = vec![];
+        let mut temp_places: Vec<StateAddr> = vec![];
+        if !params.offsets_aggregate_states.is_empty() {
```

Comment:
replace with `assert!`

Reply:
Don't quite understand why `assert!` need be used here

Reply:
When the aggregate function is empty, we will not execute this part. So in this case, this is a bug.

Reply:
Originally, only when the group by is empty will go here, if the aggregation function is also empty, then certainly will not go here, if it comes is a bug. Thanks for your explanation!